src/main/java/karuna/karuna_backend/Authentication/Utils/SecurityContextUtil.java

```diff
@@ -0,0 +1,44 @@
+package karuna.karuna_backend.Authentication.Utils;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.Assert;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class to manipulate the Spring Security Context.
+ * Provides methods to programmatically set the security context using a username and roles.
+ */
+public class SecurityContextUtil {
```

Comment:
Add @NoArgsConstructor with private access if u have only static methods, we need to proctect creating objects with this class